### PR TITLE
Generics refactor

### PR DESCRIPTION
**Pull request by worryg0d**
The PR provides refactor of generic fields of the cluster resources

### CODE REVIEW

**Review by ribaraka — CHANGES_REQUESTED** [review]

File: apis/clusters/v1beta1/generic_spec.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	// Name [ 3 .. 32 ] characters.
+	Name string `json:"name,omitempty"`
+
+	Version string `json:"version,omitempty"`
+
+	// The PCI compliance standards relate to the security of user data and transactional information.
+	// Can only be applied clusters provisioned on AWS_VPC, running Cassandra, Kafka, Elasticsearch and Redis.
+	PCICompliance bool `json:"pciCompliance,omitempty"`
+
+	PrivateNetworkCluster bool `json:"privateNetworkCluster,omitempty"`
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply apis/clusters/v1beta1/generic_spec.go]: Removed

File: apis/clusters/v1beta1/generic_spec.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+	return true
+}
+
+func (s *GenericClusterSpec) FromInstAPI(model *models.GenericClusterFields) {
+	s.Name = model.Name
+	s.PCICompliance = model.PCIComplianceMode
+	s.PrivateNetworkCluster = model.PrivateNetworkCluster
+	s.SLATier = model.SLATier
+	s.Description = model.Description
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply apis/clusters/v1beta1/generic_spec.go:56]: Fixed

File: apis/clusters/v1beta1/generic_spec.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+			models.ClusterNameRegExp)
+	}
+
+	if len(s.TwoFactorDelete) > 1 {
+		return fmt.Errorf("two factor delete should not have more than 1 item")
+	}
+
+	if s.Description != "" {
+		return fmt.Errorf("description is not supported yet, when create a cluster. You can add this field when the cluster is in the running state")
+	}
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply apis/clusters/v1beta1/generic_spec.go]: Removed

File: apis/clusters/v1beta1/generic_spec.go [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+	return nil
+}
+
+func (s *GenericClusterSpec) ClusterSettingsNeedUpdate(iCluster *GenericClusterSpec) bool {
+	return len(s.TwoFactorDelete) != 0 && len(iCluster.TwoFactorDelete) == 0 ||
+		s.Description != iCluster.Description
+}
+
+func (s *GenericClusterSpec) ClusterSettingsUpdateToInstAPI() *models.ClusterSettings {
+	settingsToAPI := &models.ClusterSettings{}
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply apis/clusters/v1beta1/generic_spec.go]: Done

File: controllers/clusters/opensearch_controller.go [hunk 5]
```diff
@@ -109,141 +111,166 @@ func (r *OpenSearchReconciler) Reconcile(ctx context.Context, req ctrl.Request)
...
-		if err != nil {
-			logger.Error(err, "Cannot start OpenSearch cluster backups check job",
-				"cluster ID", o.Status.ID)
+	return nil
+}
 
-			r.EventRecorder.Eventf(o, models.Warning, models.CreationFailed,
-				"Cluster backups check job is failed. Reason: %v", err)
+func (r *OpenSearchReconciler) startClusterJobs(o *v1beta1.OpenSearch) error {
+	err := r.startClusterStatusJob(o)
```
Comment (ribaraka) [inline controllers/clusters/opensearch_controller.go:227]: startClusterStatusJob -> startClusterSyncJob

File: pkg/instaclustr/client.go [hunk 6]
```diff
@@ -2597,3 +2648,64 @@ func (c *Client) DeleteOpenSearchEgressRule(id string) error {
 
 	return nil
 }
+
+func (c *Client) CreateOpenSearchCluster(url string, spec, dst any) error {
```
Comment (ribaraka) [inline pkg/instaclustr/client.go:2652]: delete it

**Review by ribaraka — CHANGES_REQUESTED** [review]

File: pkg/instaclustr/client.go (hunk 6 above)
Comment (worryg0d) [reply pkg/instaclustr/client.go]: Done

File: controllers/clusters/opensearch_controller.go (hunk 5 above)
Comment (worryg0d) [reply controllers/clusters/opensearch_controller.go]: Done

File: apis/clusters/v1beta1/opensearch_types.go [hunk 7]
```diff
@@ -203,19 +201,15 @@ func (oss *OpenSearchSpec) ToInstAPI() *models.OpenSearchCluster {
...
-				Name:                dc.Name,
-				Network:             dc.Network,
-				AWSSettings:         providerSettings.AWSSettings,
-				GCPSettings:         providerSettings.GCPSettings,
-				AzureSettings:       providerSettings.AzureSettings,
-				Tags:                tagsToInstAPI(dc.Tags),
-				CloudProvider:       dc.CloudProvider,
-				Region:              dc.Region,
-				ProviderAccountName: dc.ProviderAccountName,
+				Name:                  dc.Name,
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply apis/clusters/v1beta1/opensearch_types.go]: Done

File: Makefile [hunk 8]
```diff
@@ -83,7 +83,7 @@ test-webhooks:
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./apis/clusters/v1beta1 -coverprofile cover.out
 
 .PHONY: test
- test: manifests generate fmt vet docker-build-server-stub run-server-stub envtest test-clusters test-clusterresources test-kafkamanagement test-users stop-server-stub
+ test: #manifests generate fmt vet docker-build-server-stub run-server-stub envtest test-clusters test-clusterresources test-kafkamanagement test-users stop-server-stub
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply Makefile]: reverted

**Review by ribaraka — APPROVED** [review]

**Pull request merged**